http-clients/url-connection-client/src/test/java/software/amazon/awssdk/http/urlconnection/Expect100ContinueTest.java

```diff
@@ -0,0 +1,268 @@
+            assertThat(response.httpResponse().firstMatchingHeader("x-amz-test-header")).hasValue("foo");
+        }
+    }
+
+    @Test
+    public void expect100ContinueWorksWith204() throws Exception {
+        Handler handler = new AbstractHandler() {
+            @Override
+            public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+                throws IOException {
+                response.setStatus(204);
+                response.addHeader("x-amz-test-header", "foo");
+                response.flushBuffer();
+            }
+        };
+
+        try (SdkHttpClient client = UrlConnectionHttpClient.create();
+             EmbeddedServer server = new EmbeddedServer(handler)) {
+            HttpExecuteResponse response = sendRequest(client, server);
+            assertThat(response.httpResponse().statusCode()).isEqualTo(204);
+            assertThat(response.httpResponse().firstMatchingHeader("x-amz-test-header")).hasValue("foo");
+        }
+    }
+
+    @Test
+    public void expect100ContinueWorksWith304() throws Exception {
+        Handler handler = new AbstractHandler() {
+            @Override
+            public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+                throws IOException {
+                response.setStatus(304);
+                response.addHeader("x-amz-test-header", "foo");
+                response.flushBuffer();
+            }
+        };
+
+        try (SdkHttpClient client = UrlConnectionHttpClient.create();
+             EmbeddedServer server = new EmbeddedServer(handler)) {
+            HttpExecuteResponse response = sendRequest(client, server);
+            assertThat(response.httpResponse().statusCode()).isEqualTo(304);
+            assertThat(response.httpResponse().firstMatchingHeader("x-amz-test-header")).hasValue("foo");
+        }
+    }
+
+    @Test
+    public void expect100ContinueWorksWith417() throws Exception {
+        Handler handler = new AbstractHandler() {
+            @Override
+            public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response)
+                throws IOException {
+                response.setStatus(417);
+                response.addHeader("x-amz-test-header", "foo");
+                response.flushBuffer();
+            }
+        };
+
+        try (SdkHttpClient client = UrlConnectionHttpClient.create();
+             EmbeddedServer server = new EmbeddedServer(handler)) {
+            HttpExecuteResponse response = sendRequest(client, server);
+            assertThat(response.httpResponse().statusCode()).isEqualTo(417);
+            assertThat(response.httpResponse().firstMatchingHeader("x-amz-test-header")).hasValue("foo");
+        }
+    }
+
+    @Test
+    public void expect100ContinueWorksWithZeroContentLength500() throws Exception {
```

Comment:
Nit: Should we assert that no warnings were logged?

Reply:
Is that to prevent someone upgrading the log to warn?

Reply:
I meant to assert that nothing was logged at all (if there was no missing behavior).